remove unused cruft

clockfort opened the pull request:
checkstyle told me to delete things; happy to oblige

deleted:

# KVS#TearDown

Teardown's documentation says it deletes everything and closes, which hasn't been true in a long time and was not the case in any KVS at all. (except ones like InMemoryKVS, where KVS#close() also deletes all data for this particular KVS)

Anyone actually using this API endpoint means to be calling KVS#close() and is lucky we weren't already deleting all of their data as a penalty for them not reading the javadocs (or if they meant the call to delete their data, they were also in error, because it doesn't delete the data currently...)

# KVS#GetRangeWithHistory

This was replaced a long time ago with KVS#getRangeOfTimestamps

# Partitioning / Remoting KVSs

This was an old intern project, and doesn't work / would throw Unimplemented exceptions on most of the modern KVSs that people actually use.

# Build System

Removed a container; looks like no rebalance of existing containers would actually help runtime, so we can for now just give up a container to the rest of Palantir.
We should (in a different commit) split up the ETE tests, because they're by far the longest running circle container and doubling our build latency now.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/palantir/atlasdb/1041)
<!-- Reviewable:end -->

clockfort commented on atlasdb-rocksdb-tests/src/test/java/com/palantir/atlasdb/cleaner/RocksDbSweeperTest.java:
```diff
@@ -27,10 +27,9 @@
 
-    @Override
     @After
     public void tearDown() {
```
I kept the tearDown() method name for the 3 or so tests that actually do teardown by manually cleaning up

berler commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TieredKeyValueService.java:
```diff
@@ -133,13 +133,6 @@ public void close() {
-    public void teardown() {
-        primary.teardown();
-        secondary.teardown();
-        executor.shutdown();
```
does the executor shutdown happen somewhere else now?

clockfort replied:
Yep! It happens in close() already / still.

jboreiko commented:
@gbrova is working on splitting up those tests (as you mention at the bottom of your PR message)

clockfort commented:
Cool, glad to hear that someone's already looking into fast building.

I just rebased over the merge conflict and fixed the test that two of us modified.

jboreiko commented:
cool, reviewing now

jboreiko commented:
Reviewed 11 of 28 files at r1, 87 of 102 files at r3, 4 of 106 files at r4, 102 of 102 files at r5.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[settings.gradle, line 30 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVUuzimafuu7WlwdET:-KTVUuzimafuu7WlwdEU:b-5zojbd) ([raw file](https://github.com/palantir/atlasdb/blob/6c4ec431167a3e810e2da9f990108dedef5ef010/settings.gradle#L30)):*
> ```Gradle
> include ":atlasdb-timelock-server"
> include ":atlasdb-commons"
> include ":atlasdb-partition-manager"
> ```

should be able to drop this as well

---

*[atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java, line 390 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVO6KSjfaLolYVr6er:-KTVO6KTa8iKIX4JOeTf:b-m9nl6w) ([raw file](https://github.com/palantir/atlasdb/blob/42b4595d456400240a49260f7ecec94194c810da/atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java#L390)):*
> ```Java
>     @Consumes(MediaType.APPLICATION_JSON)
>     @Idempotent
>     ClosableIterator<RowResult<Set<Value>>> getRangeWithHistory(@QueryParam("tableRef") TableReference tableRef,
> ```

Glad to see something like this go, how did you go about determining whether it was unused outside of the library.  I'm always tempted to mark these as deprecated and then remove them, but that might be process getting in the way of execution.  And just to clarify, the only difference between these two methods appears to be the consistency guarantee?  Not even in the C* one, they are the same impl except history doesn't have the switch for column batching.  Lol and it's unsupported in DBKVS

---

*[atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java, line 213 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVQLZXE4QMxuVU73sI:-KTVQLZXE4QMxuVU73sJ:b63rj73) ([raw file](https://github.com/palantir/atlasdb/blob/42b4595d456400240a49260f7ecec94194c810da/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java#L213)):*
> ```Java
>                         break;
>                     }
>                     values.add(Value.createWithCopyOfData(entry.getValue(), key.ts));
> ```

This method has the copy protection we need for inMemory, we need to move this copy protection to the getRangeOfTimestamps.  This also makes me suspect someone hits this method instead of the other because that doesn't have the copy protection.

---

*[atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java, line 104 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVRAG8ROa8Kg8CR_He:-KTVRAG8ROa8Kg8CR_Hf:b-psbbid) ([raw file](https://github.com/palantir/atlasdb/blob/6c4ec431167a3e810e2da9f990108dedef5ef010/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java#L104)):*
> ```Java
> 
>     @Override
>     public Map<RangeRequest, TokenBackedBasicResultsPage<RowResult<Value>, byte[]>> getFirstBatchForRanges(TableReference tableRef,
> ```

Did this sneak through?  seems to be the same impl as would fall through to in ForwardingKeyValueService unless I'm missing a change

---

*[atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsKeyValueServiceTest.java, line 39 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVRLoSrrKQW-TfZULp:-KTVRLoSrrKQW-TfZULq:bb6un39) ([raw file](https://github.com/palantir/atlasdb/blob/42b4595d456400240a49260f7ecec94194c810da/atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/DbkvsKeyValueServiceTest.java#L39)):*
> ```Java
>     @Override
>     @Ignore
>     public void testGetRangeWithHistory() {
> ```

Dope

---

*[docs/source/release_notes/release-notes.rst, line 46 at r5](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVfZsNieaTnaUu9XQq:-KTVfZsNieaTnaUu9XQr:b-iktemp) ([raw file](https://github.com/palantir/atlasdb/blob/1a904d1cf248b50548c59ff1db5e462f09a73631/docs/source/release_notes/release-notes.rst#L46)):*
> ```reStructuredText
>          - Removed KeyValueService.[initializeFromFresh|tearDown|getRangeWithHistory]
>            It is likely all callers of tearDown just want to call close.
>            (`Pull Request <https://github.com/palantir/atlasdb/pull/1041>`__)
> ```

Would also like to see mention of the two removed KVSs

---

*[gradle/baseline.gradle, line 14 at r5](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVfEtOznS12d5PhVtQ:-KTVfEtOznS12d5PhVtR:b-8htpg6) ([raw file](https://github.com/palantir/atlasdb/blob/1a904d1cf248b50548c59ff1db5e462f09a73631/gradle/baseline.gradle#L14)):*
> ```Gradle
>         'atlasdb-jdbc-tests',
>         'atlasdb-lock-api',
>         'atlasdb-partition-manager',
> ```

here again?

---

*[scripts/circle-ci/run-circle-tests.sh, line 19 at r5](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVfQIaaw2Z-eresX78:-KTVfQIaaw2Z-eresX79:b-5tpzig) ([raw file](https://github.com/palantir/atlasdb/blob/1a904d1cf248b50548c59ff1db5e462f09a73631/scripts/circle-ci/run-circle-tests.sh#L19)):*
> ```Shell
> 
> # Container 0 - runs tasks not found in the below containers
> CONTAINER_0_EXCLUDE=("${CONTAINER_1[@]}" "${CONTAINER_2[@]}" "${CONTAINER_3[@]}" "${CONTAINER_4[@]}"  "${CONTAINER_5[@]}")
> ```

Unfortunately a few more refs to container_5 that we might want to clean up

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/1041)*
<!-- Sent from Reviewable.io -->

jboreiko commented:
It's kinda interesting that we had all of this serialization code for atlas api objects for the remoting keyvalue service and then again for console in a much smaller form.

clockfort commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[settings.gradle, line 30 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVUuzimafuu7WlwdET:-KTVqxDQi99fHfNQ0Y2E:b-896fix) ([raw file](https://github.com/palantir/atlasdb/blob/6c4ec431167a3e810e2da9f990108dedef5ef010/settings.gradle#L30)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> should be able to drop this as well
</details>
Done.

---

*[atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java, line 390 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVO6KSjfaLolYVr6er:-KTVqQjT4diT9uAfHXAW:bwdy62s) ([raw file](https://github.com/palantir/atlasdb/blob/42b4595d456400240a49260f7ecec94194c810da/atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java#L390)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> Glad to see something like this go, how did you go about determining whether it was unused outside of the library.  I'm always tempted to mark these as deprecated and then remove them, but that might be process getting in the way of execution.  And just to clarify, the only difference between these two methods appears to be the consistency guarantee?  Not even in the C* one, they are the same impl except history doesn't have the switch for column batching.  Lol and it's unsupported in DBKVS
</details>
We've been meaning to remove it for a while. Deprecation isn't necessary, given that it straight up hasn't worked in a while and it's a really weird internals-exposing call. I also checked to make sure PG wasn't using it in some real way besides delegating KVSs.

consistency guarantee is the same, this one returns historical values that a given range used to have, whereas getRangeOfTimestamps returns just the timestamps of old versions in that range.

---

*[atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java, line 213 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVQLZXE4QMxuVU73sI:-KTVuG9IJIbyswmhitDz:bl6vi66) ([raw file](https://github.com/palantir/atlasdb/blob/42b4595d456400240a49260f7ecec94194c810da/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java#L213)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> This method has the copy protection we need for inMemory, we need to move this copy protection to the getRangeOfTimestamps.  This also makes me suspect someone hits this method instead of the other because that doesn't have the copy protection.
</details>
getRangeOfTimestamps looks okay to me, it creates a new Set<Long> and the Longs come from longs so they'll be new Objects and not dangerous refs to mutable KVS internals. (Hopefully this is the most java sentence I'll type all day)

I would not worry about it being used outside of atlas, the only reason someone would be using this is to implement something like sweep or really something more like tiered KVS or partitioning KVS or other very atlas-internal style things that operate outside of the entire transaction framework.

---

*[atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java, line 104 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVRAG8ROa8Kg8CR_He:-KTVr8ucGwhzHKwULuKt:bbzwa8w) ([raw file](https://github.com/palantir/atlasdb/blob/6c4ec431167a3e810e2da9f990108dedef5ef010/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java#L104)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> Did this sneak through?  seems to be the same impl as would fall through to in ForwardingKeyValueService unless I'm missing a change
</details>
Nope, it's abstract :turtle:s all the way down

public abstract class ForwardingObject {
  protected abstract Object delegate();
}


public abstract class ForwardingKeyValueService extends ForwardingObject implements KeyValueService {
    @Override
    protected abstract KeyValueService delegate();
}

---

*[docs/source/release_notes/release-notes.rst, line 46 at r5](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVfZsNieaTnaUu9XQq:-KTVs4wkXlqKmtcbsZBm:b-iwf4hq) ([raw file](https://github.com/palantir/atlasdb/blob/1a904d1cf248b50548c59ff1db5e462f09a73631/docs/source/release_notes/release-notes.rst#L46)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> Would also like to see mention of the two removed KVSs
</details>
Okay, will add them in. I didn't list those because I didn't feel it was a change a user would care about. 

---

*[gradle/baseline.gradle, line 14 at r5](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVfEtOznS12d5PhVtQ:-KTVs_UNOiSRbgWwjOkm:b-896fix) ([raw file](https://github.com/palantir/atlasdb/blob/1a904d1cf248b50548c59ff1db5e462f09a73631/gradle/baseline.gradle#L14)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> here again?
</details>
Done.

---

*[scripts/circle-ci/run-circle-tests.sh, line 19 at r5](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVfQIaaw2Z-eresX78:-KTVsfmzOM33hcd1PRkT:b1jxzi3) ([raw file](https://github.com/palantir/atlasdb/blob/1a904d1cf248b50548c59ff1db5e462f09a73631/scripts/circle-ci/run-circle-tests.sh#L19)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> Unfortunately a few more refs to container_5 that we might want to clean up
</details>
good catch, done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/1041)*
<!-- Sent from Reviewable.io -->

jboreiko commented:
Reviewed 4 of 4 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java, line 213 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVQLZXE4QMxuVU73sI:-KTWDxam7ZbuaCBd9Jf1:b851wpb) ([raw file](https://github.com/palantir/atlasdb/blob/42b4595d456400240a49260f7ecec94194c810da/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java#L213)):*
<details><summary><i>Previously, clockfort (Chris Lockfort) wrote…</i></summary>
> getRangeOfTimestamps looks okay to me, it creates a new Set<Long> and the Longs come from longs so they'll be new Objects and not dangerous refs to mutable KVS internals. (Hopefully this is the most java sentence I'll type all day)
> 
> I would not worry about it being used outside of atlas, the only reason someone would be using this is to implement something like sweep or really something more like tiered KVS or partitioning KVS or other very atlas-internal style things that operate outside of the entire transaction framework.
</details>
Ah good catch, didn't notice it was longs

---

*[atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java, line 104 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVRAG8ROa8Kg8CR_He:-KTibA1YIxFtKNdqzzvQ:b-8644nx) ([raw file](https://github.com/palantir/atlasdb/blob/6c4ec431167a3e810e2da9f990108dedef5ef010/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java#L104)):*
<details><summary><i>Previously, clockfort (Chris Lockfort) wrote…</i></summary>
> Nope, it's abstract :turtle:s all the way down
> 
> public abstract class ForwardingObject {
>   protected abstract Object delegate();
> }
> 
> 
> public abstract class ForwardingKeyValueService extends ForwardingObject implements KeyValueService {
>     @Override
>     protected abstract KeyValueService delegate();
> }
> 
</details>
Sorry not delegate(), getFirstBatchForRanges() in ForwardingKeyValueService should pick this up exactly as is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/1041)*
<!-- Sent from Reviewable.io -->

clockfort commented:
Review status: 112 of 113 files reviewed at latest revision, 1 unresolved discussion.

---

*[atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java, line 104 at r3](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-KTVRAG8ROa8Kg8CR_He:-KTkDf4hfp1QCNEf3Foq:b-896fix) ([raw file](https://github.com/palantir/atlasdb/blob/6c4ec431167a3e810e2da9f990108dedef5ef010/atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/SweepStatsKeyValueService.java#L104)):*
<details><summary><i>Previously, jboreiko (John Boreiko) wrote…</i></summary>
> Sorry not delegate(), getFirstBatchForRanges() in ForwardingKeyValueService should pick this up exactly as is.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/1041)*
<!-- Sent from Reviewable.io -->

tpetracca commented:
Not sure if he has his own local thing, but I think @carrino was doing stuff with partitioning KVS.

clockfort commented:
@tpetracca I talked with carrino; when he wants it he'll make his own version in his project for now. I told him the things that were outdated / duplicated in this current version so he knows what he'll need to write from scratch vs. steal

jboreiko commented:
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/1041)*
<!-- Sent from Reviewable.io -->

jboreiko commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/1041#-:-KTn8GNkoi-o_DDgR8R4:bnfp4nl)*
<!-- Sent from Reviewable.io -->
